platform/api/aws/ami.go

```diff
@@ -28,22 +28,9 @@ type releaseAMIs struct {
 	}
 }
 
-func (a *API) AMI() (resolvedAMI string, resolveAMIErr error) {
-	a.opts.resolvedAMILock.Lock()
-	defer a.opts.resolvedAMILock.Unlock()
-
-	// Use cache if we can
-	if a.opts.ResolvedAMI != "" {
-		return a.opts.ResolvedAMI, nil
-	}
-
-	// If successfully resolved, cache the AMI id for next call.
-	// Decide based on the named returns
-	defer func() {
-		if resolveAMIErr == nil {
-			a.opts.ResolvedAMI = resolvedAMI
-		}
-	}()
+func (a *API) AMI() (string, error) {
+	a.amiResolutionLock.Lock()
+	defer a.amiResolutionLock.Unlock()
```

Comment:
.. oh, yeah, it's also not doing a `describe-images` call anymore so moot point. That was the original reason for not putting it in `New`. that comment can be half ignored. Options still seems reasonable to me though.